messaging/src/main/java/org/axonframework/configuration/Module.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.configuration;
+
+/**
+ * Interface describing a module of Axon Framework's configuration API.
+ * <p>
+ * Modules are relatively independent. They can be {@link NewConfigurer#registerModule(ModuleBuilder) registered} on a
+ * parent {@link NewConfigurer} or registered in a nested style on another {@link Module} through the dedicated register
+ * module operation. Furthermore, a module is able to access the registered {@link Component Components} from the parent
+ * {@code Configurer} it is registered too. However, the parent is <b>not</b> able to retrieve components from these
+ * {@code Modules}, ensuring encapsulation.
+ *
+ * @param <M> The type of module this implementation returns. This generic allows us to support fluent interfacing.
+ * @author Allard Buijze
+ * @author Steven van Beelen
+ * @since 3.0.0
+ */
+public interface Module<M extends Module<M>> extends NewConfigurer<M> {
```

Comment:
If I read the code right, right now `Modules` can not support lifecycle operations, as the register calls throw unsupported operations.

What is the plan for the future? Are we going to support it, but is it going to delegate these calls to the parent configuration? I think we will need a way for a module to configure it

Reply:
The original intent was to enforce life cycle management to the root. However, the current approach indeed remove the option to invoke onstart/onshutdown entirely on a module, which feels off as well.
Let's deal with this in the upcoming life cycle changes PR to align, is that okay?